crates/sui-framework/packages/sui-framework/sources/random.move

```diff
@@ -122,4 +126,124 @@ module sui::random {
     ) {
         update_randomness_state(self, new_round, new_bytes, ctx);
     }
+
+
+    /// Unique randomness generator, derived from the global randomness.
+    struct RandomGenerator has drop {
+        seed: vector<u8>,
+        counter: u16,
+        buffer: vector<u8>,
+    }
+
+    /// Create a generator. Can be used to derive up to MAX_U16 * 32 random bytes.
+    public fun new_generator(r: &Random, ctx: &mut TxContext): RandomGenerator {
+        let inner = load_inner(r);
+        // TODO[crypto]: should blake be used here instead?
+        let seed = hmac_sha3_256(&inner.random_bytes,&to_bytes(fresh_object_address(ctx)));
+        RandomGenerator { seed, counter: 0, buffer: vector::empty() }
+    }
+
+    // Get the next block of random bytes.
+    fun derive_next_block(g: &mut RandomGenerator): vector<u8> {
+        g.counter = g.counter + 1;
+        // TODO[crypto]: should blake be used here instead?
+        hmac_sha3_256(&g.seed, &bcs::to_bytes(&g.counter))
+    }
+
+    // Fill the generator's buffer with 32 random bytes.
+    fun fill_buffer(g: &mut RandomGenerator) {
+        let next_block = derive_next_block(g);
+        vector::append(&mut g.buffer, next_block);
+    }
+
+    /// Generate n random bytes.
+    public fun bytes(g: &mut RandomGenerator, num_of_bytes: u16): vector<u8> {
+        // TODO[move]: should we have this limit or just leave it to gas limit?
+        assert!(num_of_bytes <= MAX_RANDOM_BYTES, ETooManyBytes);
```

Comment:
I personally think it's fine leaving it to gas. We aren't doing anything native here really so it feels fine to me